From 6c2085badea7b461245837c452a0d3d8a8c2afff Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 20 Dec 2020 22:09:28 -0800
Subject: [PATCH] ltp-pan: Use long long int to print time_t

Some newer 32bit architectures ( e.g. riscv32 ) uses 64bit time_t so
using %ld is not sufficient to print time_t, this also fixes a crash in
ltp-pan on riscv32

Upstream-Status: Submitted [https://patchwork.ozlabs.org/project/ltp/patch/20201221061415.2540216-1-raj.khem@gmail.com/]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 pan/ltp-pan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pan/ltp-pan.c b/pan/ltp-pan.c
index 8b9fbe5594..54b7cb8f26 100644
--- a/pan/ltp-pan.c
+++ b/pan/ltp-pan.c
@@ -1389,8 +1389,8 @@ static void write_test_start(struct tag_pgrp *running, int no_kmsg)
 	if (!strcmp(reporttype, "rts")) {
 
 		printf
-		    ("%s\ntag=%s stime=%ld\ncmdline=\"%s\"\ncontacts=\"%s\"\nanalysis=%s\n%s\n",
-		     "<<<test_start>>>", running->cmd->name, running->mystime,
+		    ("%s\ntag=%s stime=%lld\ncmdline=\"%s\"\ncontacts=\"%s\"\nanalysis=%s\n%s\n",
+		     "<<<test_start>>>", running->cmd->name, (long long)running->mystime,
 		     running->cmd->cmdline, "", "exit", "<<<test_output>>>");
 	}
 	fflush(stdout);
-- 
2.29.2

